core/src/main/resources/bosatsu/predef.bosatsu

```diff
@@ -127,15 +132,235 @@ def range_fold(inclusiveLower: Int, exclusiveUpper: Int, init: a, fn: a -> Int -
     a1 = fn(a, idx)
     (diff0.sub(1), a1))
 
+#############
+# String functions
+#############
 external struct String
+
 external def string_Order_fn(str0: String, str1: String) -> Comparison
 string_Order = Order(string_Order_fn)
 external def concat_String(items: List[String]) -> String
 
 external def int_to_String(i: Int) -> String
 
+external def trace(prefix: String, item: a) -> a
+
+#############
+# Support for built-in testing:
+#############
 enum Test:
   Assertion(value: Bool, message: String)
   TestSuite(name: String, tests: List[Test])
 
-external def trace(prefix: String, item: a) -> a
+#############
+# AVL trees used to implement dictionaries
```

Comment:
this code is duplicative with AvlTree.bosatsu, but debugging the predef is very painful since we parse and compile it once on initialization. Having it copied makes it easier to have tests in another file.